texastribune/data-visuals-create
# Add an optional starter to features using shared Texas Tribune style library

### What does this do?
Adds our shared style system as a dependency and an alternative _feature_ template referencing those styles.

_Important limitations to note for this first iteration_
- This really only applies to a feature for now. The new styles are available for graphics, but I'm not as familiar with the process for creating those so I didn't add anything out of the box.
- All the current templates and styles are left unchanged. The queso helpers are there if you want to go that route, but if you're under a deadline and need to use what's familiar, you could essentially ignore these changes.

### Why are we doing this?
The hope is to offer a more dry approach to how we style standard site elements, like footers, links, typography etc. This brings consistency for our readers and a centralized location for updating styles. This shared-style framework is also actively [documented](https://texastribune.github.io/queso-ui/), which could be helpful for new team members. Maybe someday soon, we could add some data viz specific elements i.e. graphics and tables documentation there.


### Added
- `/feature/app/index-queso.html` - Starter template for queso CSS framework
- `/feature/app/styles/main-queso.scss` - Starter styles for CSS framework
- `/__common__/app/styles/_typography-queso.scss` - Extra typography styles not accounted for in queso-ui and data viz specific overrides of existing helpers. Appended `-queso` as to not be confused with current _typography.scss
- `/feature/app/styles/components/_navbar.scss` - Overrides for queso navbar
- `feature/app/styles/layout/_container.scss` - Additional container classes
- `/feature/app/templates/components/navbar.html` - Standard navbar component (used on TT)
- `/feature/app/templates/components/share.html` - Standard share component (used on TT)
- `/feature/app/templates/includes/logo.html` - Standard logo include (used on TT)


### Changed
- `_package.json` - Added queso dependency
- `_variables.scss` - Added variables required for queso
- `/feature/app/scripts/components/RelatedContent.js` and `/feature/app/scripts/components/Story.js` - Sprinkled in queso helpers (to be compatible in either starter template)
- `/feature/app/styles/components/_ads.scss` - Updated teal to match TT site (it's a random darkened-teal for accessibility)
- `/feature/app/styles/main.scss` - Removed sass-mq because it comes with the queso imports and added just the base queso variables and tools. This doesn't actually add any CSS so it's helpful if you ever want to include queso helpers in this file in your project.
- `/feature/app/templates/base.html` - Added more blocks so that the queso starter template could vary in those parts. Example: Google fonts isn't needed in the queso started so we override that with an empty block.

It shows that I changed `processors.html`, but it's just that my code editor strips trailing whitespace on save so I just saved that file at some point.


-----------------------------------------------------------------------------------------------

### Testing it:
- Clone a local version of `data-visuals-create`, checkout the branch `add-queso`, and run `your/path/to/data-visuals-create/bin/data-visuals-create feature my-test-feature`
1. Run `npm start`
2. Confirm that the normal root index.html page looks unchanged. The only exception here is that darker teal in the ad text.
3. Navigate to http://localhost:3000/index-queso/. Ensure that it loads properly and looks mostly similar* to the original.

*There are deviations, but I wanted to check in about what are acceptable differences and what we should try to correct. See questions below.



###  Questions:
@mandicai /@darlacameron

1. Is it confusing to have these two coexist for now? We may want to add a reminder to delete the extra starter depending on which one gets used.
2. The most noticeable differences are in the navbar (specifically the button and padding) and the story blocks in the related content. Those I kept because they're more standard to the system, but I'm open to further customizing those sections if we need to.
3. Is there anything I should add for graphics? I kept this pretty specific to features for now just because of my own ignorance regarding the graphics workflow.
